lib/pbench/test/unit/server/conftest.py

```diff
@@ -491,6 +492,34 @@ def pbench_token(client, server_config):
     return data["auth_token"]
 
 
+@pytest.fixture()
+def attach_dataset(pbench_token, create_user):
+    """
+    Mock a Dataset attach call to return an object. We mock the Dataset.attach
+    method to avoid DB access here, however the user authentication mechanism
+    is not yet mocked so we have to look up User data.
+
+    Args:
+        pbench_token: create a "drb" user for testing
+        create_user: create a "test" user
+    """
+    datasets = {}
+    datasets["drb"] = Dataset(
+        owner="drb",  # Created by pbench_token fixture
+        controller="node",
+        name="drb",
+        access="private",
+    )
+    datasets["drb"].add()
+    datasets["test"] = Dataset(
+        owner="test",  # Created by create_user fixture
+        controller="node",
+        name="test",
+        access="private",
+    )
+    datasets["test"].add()
```

Comment:
This is ugly, since the `dict` is now completely unused. In #2417 I've reduced this to:

```
    Dataset(owner="drb", controller="node", name="drb", access="private").add()
    Dataset(owner="test", controller="node", name="test", access="private").add()
```

I'm not suggesting you have to do this now; although, if you choose to, it'll look nicer and it'll simplify my rebase. 🤞🏻

Reply:
yep, thats a good point. pushed another commit to fix it